From: Steve Dickson <SteveD@redhat.com>
Date: Tue, 9 Nov 2010 15:21:37 -0500
Subject: [fs] nfs: fix a referral error Oops
Message-id: <1289316097-1745-2-git-send-email-steved@redhat.com>
Patchwork-id: 29076
O-Subject: [RHEL5.4-Snap1 PATCH 1/1] Fix a referral error Oops
Bugzilla: 556886
RH-Acked-by: Jeff Layton <jlayton@redhat.com>

Adding a error check after the nfs4_set_client() call
in nfs4_create_referral_server() stop the oops.

Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/fs/nfs/client.c b/fs/nfs/client.c
index 1b61e86..aa5cfa2 100644
--- a/fs/nfs/client.c
+++ b/fs/nfs/client.c
@@ -1070,6 +1070,9 @@ struct nfs_server *nfs4_create_referral_server(struct nfs_clone_mount *data,
 			parent_client->retrans_timeo,
 			parent_client->retrans_count);
 
+	if (error < 0)
+		goto error;
+
 	/* Initialise the client representation from the parent server */
 	nfs_server_copy_userdata(server, parent_server);
 	server->caps |= NFS_CAP_ATOMIC_OPEN;
